File: src/test/java/com/nicomincuzzi/frameworkless/maze/NavigationTest.java
package com.nicomincuzzi.frameworkless.maze;

import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import static java.util.Collections.emptyList;
import static java.util.Collections.singletonList;
import static org.junit.jupiter.api.Assertions.assertTrue;

class NavigationTest {

    @Test
    void showResultRetroRoutePuzzle() {

        Navigation navigation = new Navigation(emptyList(), singletonList(new Room()));

        navigation.showResultRetroRoutePuzzle(new HashMap<>());
    }

    @Test
    void oneItemsIsFoundWithSuccess() {
        List<String> findingItems = singletonList("Knife");
        Navigation navigation = new Navigation(findingItems, singletonList(new Room()));

        Room roomMaze = new Room();
        Utensil utensil = new Utensil();
        utensil.setName("Knife");
        roomMaze.setObjects(singletonList(utensil));

        Map<String, GameResult> result = navigation.searchItemsMaze(roomMaze);

        GameResult expectedGameResult = result.entrySet().iterator().next().getValue();
        assertTrue(expectedGameResult.getItems().contains("Knife"));
    }

    @Test
    void noItemsIsFound() {
        Navigation navigation = new Navigation(emptyList(), singletonList(new Room()));

        Room roomMaze = new Room();
        Utensil utensil = new Utensil();
        utensil.setName("Knife");
        roomMaze.setObjects(singletonList(utensil));

        Map<String, GameResult> result = navigation.searchItemsMaze(roomMaze);

        GameResult expectedGameResult = result.entrySet().iterator().next().getValue();
        assertTrue(expectedGameResult.getItems().contains("None"));
    }
}

File: src/test/kotlin/com/nicomincuzzi/frameworkless/maze/NavigationTest.kt
package com.nicomincuzzi.frameworkless.maze

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.mockito.Mockito.mock

class NavigationTest {

    @Test
    fun showResultRetroRoutePuzzle() {
        val navigation = Navigation(emptyList(), listOf(Room()), mock(JsonManagerMaze::class.java))
        navigation.showResultRetroRoutePuzzle(HashMap())
    }

    @Test
    fun oneItemsIsFoundWithSuccess() {
        val findingItems = listOf("Knife")
        val navigation = Navigation(findingItems, listOf(Room()), mock(JsonManagerMaze::class.java))
        val utensil = Utensil(name = "Knife")
        val roomMaze = Room(objects = listOf(utensil))

        val result = navigation.searchItemsMaze(roomMaze)
        val expectedGameResult = result.entries.iterator().next().value
        Assertions.assertTrue(expectedGameResult.items.contains("Knife"))
    }

    @Test
    fun noItemsIsFound() {
        val navigation = Navigation(emptyList(), listOf(Room()), mock(JsonManagerMaze::class.java))
        val utensil = Utensil(name = "Knife")
        val roomMaze = Room(objects = listOf(utensil))

        val result = navigation.searchItemsMaze(roomMaze)
        val expectedGameResult = result.entries.iterator().next().value
        Assertions.assertTrue(expectedGameResult.items.contains("None"))
    }
}